src/arrayMethodSort.js

```diff
@@ -5,7 +5,33 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    if (compareFunction && typeof compareFunction === 'function') {
+      for (let i = 0; i < this.length - 1; i++) {
+        for (let j = 0; j < this.length - 1 - i; j++) {
+          if (compareFunction(this[j], this[j + 1]) > 0) {
+            const temp = this[j];
+
+            this[j] = this[j + 1];
+            this[j + 1] = temp;
+          }
+        }
+      }
+
+      return this;
+    } else {
```

Comment:
`else` is not needed when returning inside the previous if statement (early return)